core/src/main/java/org/apache/brooklyn/core/location/BasicLocationRegistry.java

```diff
@@ -372,7 +377,10 @@ public final Location getLocationManaged(String spec, Map locationFlags) {
 
             if (resolver != null) {
                 try {
-                    return (Maybe) Maybe.of(resolver.newLocationSpecFromString(spec, locationFlags, this));
+                    LocationSpec<? extends Location> specO = resolver.newLocationSpecFromString(spec, locationFlags, this);
```

Comment:
I feel like the naming in this function has gotten a bit lost.  We have a spec and a specO and it took me a bit too much time to understand what I was seeing.  Can't think of a better naming convention now.